### ISSUE TITLE
Explain doesn't show GPU 

### ISSUE THREAD

**Issue by eloufirhatim** [heterodb/pg-strom#379]
Hello,

When executing the EXPLAIN function on my query, I don't see the GroupAggregate or GpuPreAgg, and I followed all the steps of installation and configuration.

```
intranet_dashboard=# EXPLAIN analyze select "etl"."uniqid", count("etl"."id") from dash_etl_tb etl join dash_job_tb job on "job"."refEtl" = "etl"."id" group by "etl"."id";
                                                      QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------
 HashAggregate  (cost=13.48..13.49 rows=1 width=26) (actual time=0.021..0.021 rows=0 loops=1)
   Group Key: etl.id
   ->  Nested Loop  (cost=0.00..13.04 rows=90 width=18) (actual time=0.019..0.019 rows=0 loops=1)
         Join Filter: (etl.id = job."refEtl")
         ->  Seq Scan on dash_etl_tb etl  (cost=0.00..1.01 rows=1 width=18) (actual time=0.012..0.012 rows=1 loops=1)
         ->  Seq Scan on dash_job_tb job  (cost=0.00..10.90 rows=90 width=4) (actual time=0.004..0.004 rows=0 loops=1)
 Planning time: 0.492 ms
 Execution time: 0.100 ms
(8 lignes)

```

P.S. I am new on postgresql and all of this, if I need to give more information about something please feel free to mention it and I will, thnks :)

**Comment by eloufirhatim** [comment]
I tested another query (see below)

`
                                                                                                                        QUERY PLAN


---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------
 Group  (cost=4203.01..4204.69 rows=42 width=150)
   Group Key: etl_5b6023a6568da_projet_tb.nom, etl_5b6023a6568da_article_tb.lgn_pos_art, etl_5b6023a6568da_activite_tb.description, etl_5b6023a6568da_employee.emp_matricule, etl_5b6023a6568
da_employee.emp_lastname, etl_5b6023a6568da_employee.emp_firstname, etl_5b6023a6568da_jours_ouvrable_weekly_tb.annee, etl_5b6023a6568da_jours_ouvrable_weekly_tb.mois, etl_5b6023a6568da_jour
s_ouvrable_weekly_tb.week, etl_5b6023a6568da_jours_ouvrable_weekly_tb.jours_ouvrables, etl_5b6023a6568da_ts_bis_tb.j1, etl_5b6023a6568da_ts_bis_tb.j2, etl_5b6023a6568da_ts_bis_tb.j3, etl_5b
6023a6568da_ts_bis_tb.j4, etl_5b6023a6568da_ts_bis_tb.j5
   ->  Sort  (cost=4203.01..4203.12 rows=42 width=118)
         Sort Key: etl_5b6023a6568da_projet_tb.nom, etl_5b6023a6568da_article_tb.lgn_pos_art, etl_5b6023a6568da_activite_tb.description, etl_5b6023a6568da_employee.emp_matricule, etl_5b6023
a6568da_employee.emp_lastname, etl_5b6023a6568da_employee.emp_firstname, etl_5b6023a6568da_jours_ouvrable_weekly_tb.annee, etl_5b6023a6568da_jours_ouvrable_weekly_tb.mois, etl_5b6023a6568da
_jours_ouvrable_weekly_tb.week, etl_5b6023a6568da_jours_ouvrable_weekly_tb.jours_ouvrables, etl_5b6023a6568da_ts_bis_tb.j1, etl_5b6023a6568da_ts_bis_tb.j2, etl_5b6023a6568da_ts_bis_tb.j3, e
tl_5b6023a6568da_ts_bis_tb.j4, etl_5b6023a6568da_ts_bis_tb.j5
         ->  Hash Join  (cost=2765.86..4201.88 rows=42 width=118)
               Hash Cond: (etl_5b6023a6568da_article_tb.ref_projet = etl_5b6023a6568da_projet_tb.id_projet)
               ->  Hash Join  (cost=2732.26..4167.71 rows=42 width=98)
                     Hash Cond: (etl_5b6023a6568da_activite_tb.id_article = etl_5b6023a6568da_article_tb.id_article)
                     ->  Hash Join  (cost=2655.68..4090.55 rows=42 width=80)
                           Hash Cond: ((etl_5b6023a6568da_jours_ouvrable_weekly_tb.ref_ressource = etl_5b6023a6568da_employee.emp_number) AND (etl_5b6023a6568da_jours_ouvrable_weekly_tb.ann
ee = etl_5b6023a6568da_ts_bis_tb.annee) AND (etl_5b6023a6568da_jours_ouvrable_weekly_tb.mois = etl_5b6023a6568da_ts_bis_tb.mois) AND (etl_5b6023a6568da_jours_ouvrable_weekly_tb.week = etl_5
b6023a6568da_ts_bis_tb.num_semaine))
                           ->  Seq Scan on etl_5b6023a6568da_jours_ouvrable_weekly_tb  (cost=0.00..650.15 rows=39215 width=19)
                           ->  Hash  (cost=2639.28..2639.28 rows=820 width=90)
                                 ->  Hash Join  (cost=2250.17..2639.28 rows=820 width=90)
                                       Hash Cond: (etl_5b6023a6568da_ressources_activites_tb.ref_ressource = etl_5b6023a6568da_employee.emp_number)
                                       ->  Hash Join  (cost=2235.62..2601.80 rows=1667 width=66)
                                             Hash Cond: ((etl_5b6023a6568da_ressources_activites_tb.ref_activite = etl_5b6023a6568da_activite_tb.id_activite) AND (etl_5b6023a6568da_ressourc
es_activites_tb.ref_ressource = etl_5b6023a6568da_ts_bis_tb.id_ressource))
                                             ->  Seq Scan on etl_5b6023a6568da_ressources_activites_tb  (cost=0.00..110.89 rows=7189 width=9)
                                             ->  Hash  (cost=1581.98..1581.98 rows=43576 width=69)
                                                   ->  Hash Join  (cost=112.05..1581.98 rows=43576 width=69)
                                                         Hash Cond: (etl_5b6023a6568da_ts_bis_tb.id_activite = etl_5b6023a6568da_activite_tb.id_activite)
                                                         ->  Seq Scan on etl_5b6023a6568da_ts_bis_tb  (cost=0.00..870.76 rows=43576 width=43)
                                                         ->  Hash  (cost=65.91..65.91 rows=3691 width=26)
                                                               ->  Seq Scan on etl_5b6023a6568da_activite_tb  (cost=0.00..65.91 rows=3691 width=26)
                                       ->  Hash  (cost=8.69..8.69 rows=469 width=24)
                                             ->  Seq Scan on etl_5b6023a6568da_employee  (cost=0.00..8.69 rows=469 width=24)
                     ->  Hash  (cost=45.15..45.15 rows=2515 width=26)
                           ->  Seq Scan on etl_5b6023a6568da_article_tb  (cost=0.00..45.15 rows=2515 width=26)
               ->  Hash  (cost=19.93..19.93 rows=1093 width=28)
                     ->  Seq Scan on etl_5b6023a6568da_projet_tb  (cost=0.00..19.93 rows=1093 width=28)
(29 lignes)

`

It contains **43576** rows estimated, it's always small? Can I change the settings of pg_strom to always choose GPU-based query execution plan?

Thanks in advance

**Comment by kaigai** [comment]
43576 rows looks to me too small.
PG-Strom is designed to process at least million rows, often billion rows.

If you set `pg_strom.gpu_setup_cost=0`, optimizer may ignore the cost for device initialization.

**Comment by eloufirhatim** [comment]
Thanks a lot kaigai for your time and for the quick replies.
First I tested the pg_strom with your **testdb_init** sample database, and it's shows that the GPU is used when I execute the EXPLAIN function.
Then I set the `pg_strom.gpu_setup_cost=0` then tested my data (the 43576 rows) and that uses also GPU

You have done a really good work there :)
Thanks a lot

**Comment by baretomas** [comment]
Sorry for opening a closed issue, but I tried this on my setup and gpu does not get activated.
Only way to get GPU to be activated on queries is to raise the amount of rows to the millions for me. 

I also tried to set (both together and individually) with same result:

set gpu_operator_cost = 0;
set pg_strom.gpu_setup_cost = 0;

They dont seem to make any difference to when the GPU is activated or not.
